File: services/core/internal/model/embedding.go
package model

// EmbeddingProvider represents a configured embedding provider with OpenAI-compatible API.
type EmbeddingProvider struct {
	ProviderBase
	Dimensions int `json:"dimensions"`
}

// EmbeddingProviderResponse is the safe representation returned in API responses.
type EmbeddingProviderResponse struct {
	ProviderResponseBase
	Dimensions int `json:"dimensions"`
}

// ToResponse converts EmbeddingProvider to EmbeddingProviderResponse, hiding the API key.
func (p *EmbeddingProvider) ToResponse() EmbeddingProviderResponse {
	return EmbeddingProviderResponse{
		ProviderResponseBase: p.ToResponseBase(),
		Dimensions:           p.Dimensions,
	}
}

// EmbeddingProviderCreateRequest is the payload for creating a new embedding provider.
type EmbeddingProviderCreateRequest struct {
	ProviderCreateRequestBase
	Dimensions int `json:"dimensions"`
}

// EmbeddingProviderUpdateRequest is the payload for updating an existing embedding provider.
type EmbeddingProviderUpdateRequest struct {
	ProviderUpdateRequestBase
	Dimensions *int `json:"dimensions"`
}

// EmbeddingProviderListResponse wraps a list of embedding providers.
type EmbeddingProviderListResponse struct {
	Providers []EmbeddingProviderResponse `json:"providers"`
	Total     int                         `json:"total"`
}

// EmbeddingTestResponse is the result of a connectivity test.
type EmbeddingTestResponse struct {
	OK         bool   `json:"ok"`
	Dimensions int    `json:"dimensions,omitempty"`
	LatencyMs  int64  `json:"latency_ms,omitempty"`
	Error      string `json:"error,omitempty"`
}

File: clients/console/src/types/embedding.ts
export interface EmbeddingProvider {
  id: string
  name: string
  base_url: string
  model: string
  dimensions: number
  is_default: boolean
  is_enabled: boolean
  created_at: string
  updated_at: string
}

export interface EmbeddingProviderCreateRequest {
  name: string
  base_url: string
  api_key: string
  model: string
  dimensions?: number
  is_default?: boolean
}

export interface EmbeddingProviderUpdateRequest {
  name?: string
  base_url?: string
  api_key?: string
  model?: string
  dimensions?: number
  is_default?: boolean
  is_enabled?: boolean
}

export interface EmbeddingProviderListResponse {
  providers: EmbeddingProvider[]
  total: number
}

export interface EmbeddingTestResponse {
  ok: boolean
  dimensions?: number
  latency_ms?: number
  error?: string
}
